### PR TITLE
add resource dms rabbitmq instance

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add resource dms rabbitmq instance

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
ralated #1052 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccDmsRabbitmqInstances'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccDmsRabbitmqInstances -timeout 360m -parallel 4
=== RUN   TestAccDmsRabbitmqInstances_basic
=== PAUSE TestAccDmsRabbitmqInstances_basic
=== RUN   TestAccDmsRabbitmqInstances_withEpsId
=== PAUSE TestAccDmsRabbitmqInstances_withEpsId
=== CONT  TestAccDmsRabbitmqInstances_basic
=== CONT  TestAccDmsRabbitmqInstances_withEpsId
--- PASS: TestAccDmsRabbitmqInstances_withEpsId (874.64s)
--- PASS: TestAccDmsRabbitmqInstances_basic (875.15s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       875.261s
```

### CODE REVIEW

File: docs/resources/dms_rabbitmq_instance.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+* `region` - (Optional, String, ForceNew) The region in which to create the DMS rabbitmq instance resource.
+  If omitted, the provider-level region will be used. Changing this creates a new instance resource.
+
+* `name` - (Required, String) Specifies the name of the DMS rabbitmq instance. An instance name starts with a letter,
+    consists of 4 to 64 characters, and supports only letters, digits, hyphens (-) and underscores (_).
+
+* `description` - (Optional, String) Specifies the description of the DMS rabbitmq instance. It is a character
+    string containing not more than 1024 characters.
+
+* `engine_version` - (Required, String, ForceNew) Specifies the version of the rabbitmq engine.
```
Comment (ShiChangkuo) [inline docs/resources/dms_rabbitmq_instance.md]: suggest to set it be optional and defaults to "3.7.17"

File: docs/resources/dms_rabbitmq_instance.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    Changing this creates a new instance resource.
+
+* `storage_space` - (Required, Int, ForceNew) Specifies the message storage space. Value range:
+    - Single-node RabbitMQ instance: 100–90000 GB
+    - Cluster RabbitMQ instance: 100 GB x Number of nodes to 90000 GB, 200 GB x Number of nodes to 90000 GB, and 300 GB x Number of nodes to 90000 GB
+    
+    Changing this creates a new instance resource.
+
+* `storage_spec_code` - (Required, String, ForceNew) Specifies the storage I/O specification. Value range:
+    - dms.physical.storage.normal
```
Comment (ShiChangkuo) [inline docs/resources/dms_rabbitmq_instance.md]: seems "dms.physical.storage.normal" is not supportted in the console, maybe sold out.

File: docs/resources/dms_rabbitmq_instance.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    Changing this creates a new instance resource.
+
+* `password` - (Required, String, ForceNew) Specifies the password of the DMS rabbitmq instance. A password 
+    must meet the following complexity requirements: Must be 8 to 32 characters long.
+    Must contain at least 2 of the following character types: lowercase letters, uppercase
+	letters, digits, and special characters (`~!@#$%^&*()-_=+\|[{}]:'",<.>/?).
+    Changing this creates a new instance resource.
+
+* `maintain_begin` - (Optional, String) Specifies the time at which a maintenance time window starts.
+    Format: HH:mm:ss.
```
Comment (ShiChangkuo) [inline docs/resources/dms_rabbitmq_instance.md]: the format should be **HH:mm**

File: docs/resources/dms_rabbitmq_instance.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    The start time and end time of a maintenance time window must indicate the time segment of
+	a supported maintenance time window. The start time must be set to 22:00, 02:00, 06:00, 10:00,
+    14:00, or 18:00. Parameters `maintain_begin` and `maintain_end` must be set in pairs. If
+    parameter `maintain_begin` is left blank, parameter `maintain_end` is also blank. In this case, 
+    the system automatically allocates the default start time 02:00.
+
+* `maintain_end` - (Optional, String) Specifies the time at which a maintenance time window ends.
+    Format: HH:mm:ss.
```
Comment (ShiChangkuo) [inline docs/resources/dms_rabbitmq_instance.md]: the format should be **HH:mm**

File: huaweicloud/resource_huaweicloud_dms_rabbitmq_instance.go [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+	config := meta.(*config.Config)
+	dmsV2Client, err := config.DmsV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating HuaweiCloud dms instance client: %s", err)
+	}
+
+	_, err = instances.Get(dmsV2Client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "instance")
+	}
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_dms_rabbitmq_instance.go]: it's unnecessary to Get the instance before deleting it.

File: huaweicloud/resource_huaweicloud_dms_rabbitmq_instance_test.go [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  engine        = "rabbitmq"
+  instance_type = "cluster"
+  version       = "3.7.17"
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name        = "%s"
+  description = "secgroup for rabbitmq"
+}
+
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_dms_rabbitmq_instance_test.go:161]: suggest to wrap L135-L154 into a base config function.

File: docs/resources/dms_rabbitmq_instance.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/dms_rabbitmq_instance.md]: fixed

File: docs/resources/dms_rabbitmq_instance.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/dms_rabbitmq_instance.md]: fixed

File: docs/resources/dms_rabbitmq_instance.md (hunk 4 above)
Comment (Jason-Zhang9309) [reply docs/resources/dms_rabbitmq_instance.md]: fixed

File: docs/resources/dms_rabbitmq_instance.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/dms_rabbitmq_instance.md]: fixed

File: huaweicloud/resource_huaweicloud_dms_rabbitmq_instance.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_dms_rabbitmq_instance.go]: fixed

File: huaweicloud/resource_huaweicloud_dms_rabbitmq_instance_test.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_dms_rabbitmq_instance_test.go:161]: fixed

**Pull request merged**